airbyte-webapp/src/components/GeographyDropdown/GeographyDropdown.tsx

```diff
@@ -0,0 +1,94 @@
+import { faPlus } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import React, { useState } from "react";
+import { useIntl } from "react-intl";
+import { components, OptionProps, MenuListProps, StylesConfig } from "react-select";
+
+import "/node_modules/flag-icons/css/flag-icons.min.css";
+
+import { DropDown } from "components/ui/DropDown";
+import { DropdownProps } from "components/ui/DropDown";
+
+import { Geography } from "core/request/AirbyteClient";
+import { theme } from "packages/cloud/theme";
+
+import styles from "./GeographyDropdown.module.scss";
+
+const GeographyOption: React.FC<OptionProps> = (props) => {
+  const { formatMessage } = useIntl();
+  return (
+    <components.Option className={styles.option} {...props}>
+      <div className={styles.flag}>
+        <span className={`fi fi-${props.label === "auto" ? "us" : props.label}`} />
+      </div>
+      <span className={styles.label}>{formatMessage({ id: `geography.${props.label}` })}</span>
+    </components.Option>
+  );
+};
+
+const MenuList = (props: MenuListProps) => {
+  const { formatMessage } = useIntl();
+  return (
+    <components.MenuList {...props}>
+      {props.children}
+      <div className={styles.menuList}>
+        <FontAwesomeIcon icon={faPlus} />
+        <span>{formatMessage({ id: "geography.new" })}</span>
+      </div>
+    </components.MenuList>
+  );
+};
+
+interface GeographySelectOption {
+  label: Geography;
+  value: Geography;
+}
+
+export const GeographyDropdown: React.FC<DropdownProps<GeographySelectOption>> = ({ options }) => {
+  const [option, setOption] = useState();
+  const { formatMessage } = useIntl();
+  const handleOptionSelect = (e: any) => {
+    setOption(e.label);
+  };
+
+  const formatOptionLabel = ({ label }: { label: string }) => {
+    return (
+      <div className={styles.optionLabel}>
+        <div className={styles.flag}>
+          <span className={`fi fi-${label === "auto" ? "us" : label}`} />
+        </div>
+        <span className={styles.label}>{formatMessage({ id: `geography.${label}` })}</span>
+      </div>
+    );
+  };
+
+  const customStyles: StylesConfig<GeographySelectOption> = {
+    option: (provided, state) => ({
+      ...provided,
+      color: "black",
+      backgroundColor: state.isFocused ? theme.blue50 : theme.white,
```

Comment:
@natalyjazzviolin how about keeping the `state.isSelected` styling you had here, but combining it with another check for `state.isFocused?`

```suggestion
      backgroundColor: state.isSelected ? theme.blue50 : state.isFocused ? theme.grey50 : theme.white,
```

Reply:
@josephkmh just saw this after pushing! I found a different way to do it that's hopefully cleaner!